app/ui-react/packages/ui/src/Data/Virtualizations/ViewEditor/PreviewResults.tsx

```diff
@@ -0,0 +1,87 @@
+import { EmptyState, EmptyStateBody, Title } from '@patternfly/react-core';
+import { Table } from 'patternfly-react';
+import * as React from 'react';
+import { PageSection } from '../../../../src/Layout';
+import { GenericTable } from '../../../Shared/GenericTable';
+
+export interface IPreviewResultsProps {
+  /**
+   * Array of column info for the query results.  (The column id and display label)
+   * Example:
+   * [ { id: 'fName', label: 'First Name'},
+   *   { id: 'lName', label: 'Last Name'},
+   *   { id: 'country', label: 'Country' }
+   * ]
+   */
+  queryResultCols: IColumn[];
+  /**
+   * Array of query result rows - must match up with column ids
+   * Example:
+   * [ { fName: 'Jean', lName: 'Frissilla', country: 'Italy' },
+   *   { fName: 'John', lName: 'Johnson', country: 'US' },
+   *   { fName: 'Juan', lName: 'Bautista', country: 'Brazil' },
+   *   { fName: 'Jordan', lName: 'Dristol', country: 'Ontario' }
+   * ]
+   */
+  queryResultRows: Array<{}>;
+  i18nEmptyResultsTitle: string;
+  i18nEmptyResultsMsg: string;
+}
```

Comment:
Should there be a `loading` property maybe for when the results are being refreshed?

Reply:
Yes, that would be a good improvement.  I'll create an issue to add the loading state

Reply:
Sounds good